omnipaxos_storage/src/persistent_storage.rs

```diff
@@ -6,14 +6,18 @@ use omnipaxos_core::{
     ballot_leader_election::Ballot,
     storage::{Entry, Snapshot, StopSign, StopSignEntry, Storage},
 };
-use rocksdb::{Options, DB};
 use serde::{Deserialize, Serialize};
-use std::{fs, iter::FromIterator, marker::PhantomData};
+use std::{iter::FromIterator, marker::PhantomData};
 use zerocopy::{AsBytes, FromBytes};
 
-const DEFAULT: &str = "/default_storage";
-const COMMITLOG: &str = "/commitlog";
-const ROCKSDB: &str = "/rocksDB";
+#[cfg(feature = "rocksdb")]
+use rocksdb::{Options, DB};
+#[cfg(not(feature = "rocksdb"))]
```

Comment:
Instead of using `#[cfg(not(feature = "rocksdb"))]` everywhere, use:
```suggestion
#[cfg(feature = "sled")]
```

And add this: https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features somewhere so an error is thrown if somebody tries to use both sled and rocks at the same time.